libs/parallelism/algorithms/tests/unit/container_algorithms/inplace_merge_range.cpp

```diff
@@ -177,16 +238,162 @@ void test_inplace_merge()
     test_inplace_merge<user_defined_type>();
 }
 
+template <typename IteratorTag, typename DataType>
+void test_inplace_merge_stable()
+{
+    ////////// Test cases for checking whether the algorithm is stable.
+    using namespace hpx::execution;
+
+    int rand_base = _gen();
+
+    ////////// Test cases for checking whether the algorithm is stable.
+    test_inplace_merge_stable(seq, IteratorTag(), int(), rand_base);
+    test_inplace_merge_stable(par, IteratorTag(), int(), rand_base);
+    test_inplace_merge_stable(par_unseq, IteratorTag(), int(), rand_base);
+    test_inplace_merge_stable(
+        seq, IteratorTag(), user_defined_type(), rand_base);
+    test_inplace_merge_stable(
+        par, IteratorTag(), user_defined_type(), rand_base);
+    test_inplace_merge_stable(
+        par_unseq, IteratorTag(), user_defined_type(), rand_base);
+}
+
+void test_inplace_merge_stable()
+{
+    test_inplace_merge_stable<std::random_access_iterator_tag, int>();
+    test_inplace_merge_stable<std::random_access_iterator_tag,
+        user_defined_type>();
+}
+
+///////////////////////////////////////////////////////////////////////////////
+template <typename IteratorTag, typename DataType>
+void test_inplace_merge_etc(IteratorTag, DataType, int rand_base)
+{
+    typedef typename std::vector<DataType>::iterator base_iterator;
+
+    std::size_t const left_size = 300007, right_size = 123456;
+    std::vector<DataType> res(left_size + right_size), sol, org;
+
+    base_iterator res_first = std::begin(res);
+    base_iterator res_middle = res_first + left_size;
+    base_iterator res_last = std::end(res);
+
+    std::generate(res_first, res_middle, random_fill(rand_base, 6));
+    std::generate(res_middle, res_last, random_fill(rand_base, 8));
+    std::sort(res_first, res_middle);
+    std::sort(res_middle, res_last);
+
+    org = sol = res;
+    base_iterator sol_first = std::begin(sol);
+    base_iterator sol_middle = sol_first + left_size;
+    base_iterator sol_last = std::end(sol);
+
+    // Test projection.
+    {
+        typedef test::test_iterator<base_iterator, IteratorTag> iterator;
+
+        sol = res = org;
+
+        DataType val;
+        hpx::ranges::inplace_merge(
+            policy, iterator(res_first), iterator(res_middle),
```

Comment:
```suggestion
            iterator(res_first), iterator(res_middle),
```
https://cdash.cscs.ch/viewBuildError.php?buildid=128319